### PR TITLE
Add marker component

### PR DESCRIPTION
**Pull request by Pessimistress**
A React equivalent to Mapbox's [Marker control](https://www.mapbox.com/mapbox-gl-js/api/#marker):
screen shot 2017-05-03 at 3 34 45 pm

Notes on using context:
- Pro: avoids destructing a long list of viewport props to pass down to all map overlay components
- Pro: the parent component that renders the Markers can be agnostic of the viewport
- Con: I believe context is still marked as an experimental feature of React
- Con: not supported by PureComponent. Though in this case it's not unreasonable to always re-render markers when the map updates, which is a very thin wrapper, and render the children of the markers as PureComponents.

### CODE REVIEW

File: examples/custom-interactions/view.js [hunk 1]
```diff
@@ -21,7 +21,7 @@
 import React, {Component} from 'react';
 import PropTypes from 'prop-types';
 import autobind from 'react-autobind';
-import {InteractiveMap, ScatterplotOverlay} from 'react-map-gl';
+import {InteractiveMap, ScatterplotOverlay, Marker} from 'react-map-gl';
```
Comment (ibgreen) [inline examples/custom-interactions/view.js]: Putting the Marker in this example is fine for now, but we might want to consider how we organize the examples, e.g:
- One example with all the overlays. 
- One example with markers/other react component etc.
- This particular example was initially created to demo breaking mapbox-gl's 60 degree pitch limit.

File: src/components/interactive-map.js [hunk 2]
```diff
@@ -27,6 +29,12 @@ export default class InteractiveMap extends PureComponent {
     autobind(this);
   }
 
+  getChildContext() {
```
Comment (ibgreen) [inline src/components/interactive-map.js]: If I understand it correctly, the downside of the context approach, compared to child prop injection, is that every component that wraps `StaticMap` needs to define it in this precise way for the `Marker`s to work? 

That seems to be a weakness as this InteractiveMap would need to be defined by apps that want to build their own event handling, which arguably is a key thing we are enabling in v3.0.0

Thoughts?

File: src/components/marker.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+export default class Marker extends Component {
+
+  render() {
+    const {longitude, latitude} = this.props;
+
+    const [x, y] = this.context.viewport.project([longitude, latitude]);
+
+    return createElement('div', {
```
Comment (ibgreen) [inline src/components/marker.js]: So a `Marker` is just a div that get x,y positioned absolutely on projected coordinate?

Any differences from/possible confusion with similar Mapbox features that need to be mentioned in documentation? (I am not fully up-to-date on the original API).

File: src/components/interactive-map.js (hunk 2 above)
Comment (Pessimistress) [inline src/components/interactive-map.js]: If you render the markers as children of the StaticMap then StaticMap will provide the context. Because we support hiding the map in this component, the markers are not nested under the map.
For the custom control scenario, we could add an optional prop "MapControl" that allows users to pass in a custom component without implementing InteractiveMap again.

File: examples/custom-interactions/view.js (hunk 1 above)
Comment (Pessimistress) [inline examples/custom-interactions/view.js]: Will do in another PR.

File: src/components/marker.js (hunk 3 above)
Comment (Pessimistress) [inline src/components/marker.js]: Using Mapbox you create a htm node using `document.createElement`, wrap it in a Marker or Popup control class, then call `.addTo(map)` (https://www.mapbox.com/mapbox-gl-js/api/#popup#setdomcontent). The default Marker in Mapbox is just an empty div. Popup is essentially a Marker with additional features such as alignment, close button etc.
